File: go/src/122.go
func maxProfit(prices []int) int {
	n := len(prices)
	if n == 1 {
		return 0
	}

	res := 0
	for i := 1; i < n; i++ {
		if prices[i-1] < prices[i] {
			res += (prices[i] - prices[i-1])
		}
	}
	return res
}

File: py3/122.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        if n <= 1:
            return 0
        
        profit = 0
        for i in range(1, n):
            if prices[i] > prices[i-1]:
                profit += prices[i] - prices[i-1]
            
        return profit